### PR TITLE
Added new CAT module

### PR DESCRIPTION
**Pull request by sverekar**
Hi Team,

This is a new CAT module for media use cases.

Spec: https://docs.google.com/document/d/1lqsu9v8RSWDtd4p8vAgkUwE5IWBxb4YX/edit. 
Note: The spec is in progress.

- CAT module is reference implementation which matches with the spec as of 13th May 2024.
- Implementation covers high level claims and customers are free to extend it to other claims in the spec as per their requirement.
- As of now the enc claim is supported using A256-GCM encryption algorithm only and users are free to adopt to new algorithms when it is available in EW crypto

### CODE REVIEW

File: delivery/media/cat/README.MD [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+- CAT module is reference implementation which matches with the spec as of 13th May 2024.
+- Implementation covers high level claims and customers are free to extend it to other claims in the spec as per their requirement.
+- As of now the enc claim is supported using A256-GCM encryption algorithm only and users are free to adopt to new algorithms when it is available in EW crypto
+
+## Limitations
+- The CAT module relies on the alg header claim defined in protected header of the CAT token. However the CAT spec does not mandate this, but this module uses the value defined in the protected header to identiy the algorithm used for token verification. The protected header also provides additional security. 
+- The module depends on CWT module (1.2.0 version) and its limitations are also extended. Refer [cwt](https://techdocs.akamai.com/edgeworkers/docs/cwt) doc to know its limitations.
+
+## Subfolder organization
+* **/apis**: CAT API documentation.
```
Comment (evan-hughes) [inline delivery/media/cat/README.MD:16]: It's called `api` (without an `s`) in the directory tree.

File: delivery/media/cat/examples/token-verification/main.js [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+      } catch(error) {
+        request.respondWith(401, {}, error.message)
+      }
+    } else {
+      request.respondWith(403, {}, 'Common access token is not found in cookie {name=\'Common-Access-Token\'} or query string {cat=<token>}')
+    }
+  }
+}
+
+// export function onClientResponse (request, response) {
```
Comment (evan-hughes) [inline delivery/media/cat/examples/token-verification/main.js:179]: Should there be an explanation why this is commented out?

**Review by evan-hughes — APPROVED** [review]

File: delivery/media/cat/examples/token-verification/main.js (hunk 2 above)
Comment (sverekar) [reply delivery/media/cat/examples/token-verification/main.js:183]: Added explanation. Thanks!

File: delivery/media/cat/README.MD (hunk 1 above)
Comment (sverekar) [reply delivery/media/cat/README.MD]: Updated. Thanks!

**Pull request merged**